### PR TITLE
one account integration

### PR DESCRIPTION
**Pull request by Kiura**
:sparkles: This PR adds the ability to log in/sign up using **One account**.
added `AuthCache`model migration to handle the authentication process through **One account**.
added `oneaccountId` to the `User` model in order to differentiate between account types which adds more control over the authentication process.

### CODE REVIEW

File: server/api/UserRoute.js [hunk 1]
```diff
@@ -1,12 +1,16 @@
 const jwt = require("jsonwebtoken");
 const request = require("request");
+const uuid = require("uuid/v4");
+const axios = require("axios");
```
Comment (razvanilin) [inline server/api/UserRoute.js]: The backend is already using `request` for making http request from the Node side. `request-promise` can be used for promises. Adding axios is redundant and adds extra complexity dependency-wise.

Comment (Kiura) [reply server/api/UserRoute.js]: ok, I will update the code!

**Comment by razvanilin** [comment]
Hey @Kiura, can you add a guide on how the one-account integration can be activated in the Docs? Including links to create the account, get api key, etc. I decided to add the integration in the open-source project so it will be important to have a guide out there for people. To add a new page in the docs, you can do the following:

* Install `vuepress` globally `npm install -g vuepress`
* Run the docs site locally `cd chartbrew && npm run docs:dev` (you'll have to restart this process when adding new files or changing the config file - see below)
* Create a new folder `oneaccount` in `/docs`
* Create a `README.md` file inside the folder with all the necessary instructions
* Update the `docs/.vuepress/config.js` with a new sidebar link (don't forget the emoji - you can be creative haha)
image

**Comment by Kiura** [comment]
> Hey @Kiura, can you add a guide on how the one-account integration can be activated in the Docs? Including links to create the account...

@razvanilin Absolutely! I will commit to this PR as soon as possible.

File: client/src/components/LoginForm.js [hunk 2]
```diff
@@ -4,13 +4,15 @@ import { connect } from "react-redux";
...
-  Form, Button, Message, Icon, Label, Item, Modal, Header, Input
+  Container, Form, Divider, Button, Message, Icon, Label, Item, Modal, Header, Input
 } from "semantic-ui-react";
 
-import { login, requestPasswordReset } from "../actions/user";
+import { login, requestPasswordReset, oneaccountAuth } from "../actions/user";
 import { addTeamMember } from "../actions/team";
 import { required, email } from "../config/validations";
 
+import { ONE_ACCOUNT_ENABLED } from "../config/settings";
```
Comment (Kiura) [inline client/src/components/LoginForm.js:14]: `Codacy/PR Quality Review` is not happy with this line:  `Missing file extension for "../config/settings"` although there is no import with extension in the codebase. Please let me know if this needs to be fixed or if you need me to improve the docs or something to be explained.

Comment (razvanilin) [reply client/src/components/LoginForm.js:14]: Don't worry about it, for some reason they had a pattern to show that as a warning. I removed it now as we definitely don't need the file extensions on imports

File: client/src/config/settings.template.js [hunk 3]
```diff
@@ -2,3 +2,4 @@ export const API_HOST = process.env.NODE_ENV === "production" ? "https://api.cha
 export const DOCUMENTATION_HOST = process.env.NODE_ENV === "production" ? "https://docs.chartbrew.com" : "http://localhost:8080";
 export const SITE_HOST = process.env.NODE_ENV === "production" ? "https://chartbrew.com" : "http://localhost:3000";
 export const APP_VERSION = process.env.REACT_APP_VERSION;
+export const ONE_ACCOUNT_ENABLED = !!process.env.REACT_APP_ONE_ACCOUNT_EXTERNAL_ID;
```
Comment (razvanilin) [inline client/src/config/settings.template.js:5]: Nice one 👍 I was thinking of suggesting something similar

**Pull request merged**